### PR TITLE
mesa 22.1.3

### PR DESCRIPTION
**Pull request by yurikoles**
Created with `brew bump-formula-pr`.

~~`resource` blocks may require updates.~~

Resources are checked for updates 2022-07-29.

### CODE REVIEW

**Comment by yurikoles** [comment]
```
Undefined symbols for architecture x86_64:
  "_dri_kms_driver_extensions", referenced from:
      _swkmsCreateNewScreen in libdri.a(dri_util.c.o)
  "_galliumdrm_driver_extensions", referenced from:
      _dri2CreateNewScreen in libdri.a(dri_util.c.o)
```

It seems that macOS support regressed, because I see the very same error on my machine.

**Comment by yurikoles** [comment]
It seems that meson doesn't like system LLVM on macOS, so I made formula to depend on it. I tested this change with previous version of the formula and I see more files being build with successful finish.

**Comment by yurikoles** [comment]
I forgot to mention [issue](https://gitlab.freedesktop.org/mesa/mesa/-/issues/6519) I filed upstream.

**Comment by yurikoles** [comment]
Thank you @cho-m, I forwarded your discoveries to upstream.

**Comment by yurikoles** [comment]
I'll rebase this PR after [mesa 22.0.5](https://github.com/Homebrew/homebrew-core/pull/102848) will be merged.

**Comment by yurikoles** [comment]
Depends on https://github.com/Homebrew/formula-patches/pull/869.

File: Formula/mesa.rb [hunk 1]
```diff
@@ -3,12 +3,19 @@ class Mesa < Formula
...
   license "MIT"
-  revision 1
   head "https://gitlab.freedesktop.org/mesa/mesa.git", branch: "main"
 
+  stable do
+    url "https://mesa.freedesktop.org/archive/mesa-22.1.4.tar.xz"
+    sha256 "670d8cbe8b72902a45ea2da68a9da4dc4a5d99c5953a926177adbce1b1640b76"
+
+    patch do
+      url "https://raw.githubusercontent.com/yurikoles/formula-patches/1a1d7d6/mesa/fix-build-on-macOS.patch"
```
Comment (carlocab) [inline Formula/mesa.rb:14]: Is there an upstream commit we can use instead?

Comment (yurikoles) [reply Formula/mesa.rb:14]: It doesn't apply cleanly on `22.1.4` tag.

**Comment by yurikoles** [comment]
Long build on linux is expected, since a lot of GUI formulae depend on mesa.

**Comment by yurikoles** [comment]
@carlocab macOS is green, could you please merge https://github.com/Homebrew/formula-patches/pull/869, so I'll resubmit patch from main repo, now it uses my fork.

**Comment by yurikoles** [comment]
It seems that we need to label it `CI-linux-self-hosted` in order to tests on Linux not time out.

File: Formula/mesa.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ class Mesa < Formula
     sha256 "670d8cbe8b72902a45ea2da68a9da4dc4a5d99c5953a926177adbce1b1640b76"
 
     patch do
-      url "https://raw.githubusercontent.com/yurikoles/formula-patches/1a1d7d6/mesa/fix-build-on-macOS.patch"
+      url "https://raw.githubusercontent.com/Homebrew/formula-patches/f0a40cf/mesa/fix-build-on-macOS.patch"
```
Comment (carlocab) [inline Formula/mesa.rb:14]:
```suggestion
      url "https://raw.githubusercontent.com/Homebrew/formula-patches/f0a40cf7d70ee5a25639b91d9a8088749a2dd04e/mesa/fix-build-on-macOS.patch"
```

**Comment by yurikoles** [comment]
All tests are green.

**Review by carlocab — COMMENTED** [review]
Has this been released? Both the website and mailing list show only 22.1.3.

**Comment by yurikoles** [comment]
Good point, this is a strange release, [tag](https://github.com/Mesa3D/mesa/releases/tag/mesa-22.1.4) was created 2 weeks ago. I'll downgrade it to 22.1.3.

**Comment by yurikoles** [comment]
Upstream GitLab is down.

**Comment by yurikoles** [comment]
All errors on Linux are due to missing dependencies bottles:
```
Error: assimp: no bottle available!
Error: allegro: no bottle available!
Error: cairomm: no bottle available!
Error: ceres-solver: no bottle available!
Error: botan: no bottle available!
```

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**